src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -1315,18 +1328,45 @@ private void OpenIndexesFromRecord(DatabaseRecord record, long raftIndex, Action
                     var sp = Stopwatch.StartNew();
 
                     addToInitLog($"Initializing auto index: `{name}`");
-                    OpenIndex(path, indexPath, exceptions, name, staticIndexDefinition: null, autoIndexDefinition: definition);
+                    OpenIndex(path, indexPath, exceptions, name, startIndex, staticIndexDefinition: null, autoIndexDefinition: definition);
 
                     if (Logger.IsInfoEnabled)
                         Logger.Info($"Initialized auto index: `{name}`, took: {sp.ElapsedMilliseconds:#,#;;0}ms");
                 }
             }
 
+            ForTestingPurposesOnly().AfterIndexesOpen?.Invoke();
+
+            switch (_documentDatabase.Configuration.Indexing.IndexStartupBehavior)
+            {
+                case IndexingConfiguration.IndexStartupBehaviorType.Delay:
+
+                    ExecuteForIndexes(GetIndexes(), index =>
+                    {
+                        try
+                        {
+                            switch (index.Status)
+                            {
+                                case IndexRunningStatus.Paused:
```

Comment:
Is that running concurrently for all indexes? Or waiting until they start?

Reply:
parallel, executeForIndexes uses Parallel.For